daemon/container_unix.go

```diff
@@ -1007,13 +1013,15 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 		}
 	}()
 
-	if err := container.updateEndpointNetworkSettings(n, ep); err != nil {
+	if err = container.updateEndpointNetworkSettings(n, ep); err != nil {
```

Comment:
@LK4D4 
I agree that naked return would also help. But by experience I noticed that from a code clarity standpoint, it is not optimal to use naked return functions bigger than an handful of lines.